### ISSUE TITLE
App crashes and shows "Page not found" after loading (with slow internet connection and App extensions above a certain size)

### ISSUE THREAD

**Issue by christianrr** [directus/directus#19456]
### Describe the Bug

We experience a weird bug where the Admin App seems to crash after loading/hydration, when App extensions are imported that exceed a certain size and internet speed is slow. A bunch of users have already reported this issue to us.

After the crash, the App shows "Page not found" and no further navigation is possible. Even a Refresh can not recover the App in this state. To make the App work again, it is necessary to reroute the App to /admin to rerun the login procedure. If the network speed is sufficient then, the App will start to work again.

What we know so far:
- **Size of extensions matters**: It has something to do with the _total_ size of all extensions (not the number of extensions). It happens reproducibly for extensions with a total size of >300 KB. Whether modules, displays or interfaces are used, does not make any difference.
- **Network speed matters**: The issue is reproducibly between 5-15 Mbit. Interestingly, it seems to work on _very_ slow connections (like "Slow 3G" in Chrome). Above 30 Mbit it works flawlessly.

The console indicates that the App is loaded, e.g. "🕓 Application Loaded: 575.746826171875 ms". No further error logs or erroneous requests appear.

Any idea, what could cause this issue? Rollup maybe?

### To Reproduce

1. Extract attached archive
It is basically an empty Directus instance including a dummy interface with 164 KB size (that is imported twice to exceed 300KB)
2. Run `npm install` and then `npx directus start`
3. Open Chrome (115.0.5790.114) and navigate to http://localhost:8055
4. Login with admin@example.com and directus1234
5. Open network inspector and create a throttling preset with 5 Mbit
6. Refresh Admin App until it shows --> Page not found

[Directus-Package](https://1drv.ms/u/s!AjkAcRBi2bQbg5RDnxKQBOdSoJ0fHw?e=ABpqDy)

### Directus Version

v10.5.3

### Hosting Strategy

Self-Hosted (Custom)

**Comment by rijkvanzanten** [comment]
I'm going to assume that the problem is that the extensions endpoint call is timing out, which in turn means that the app crashes in it's extensions initialization step 🤔

**Comment by idmansourali** [comment]
I confirm this issue. I'm struggling with the Page Not Found error when using a bundle extension that has a size of 1.5M.

**Comment by christianrr** [comment]
> I'm going to assume that the problem is that the extensions endpoint call is timing out, which in turn means that the app crashes in it's extensions initialization step 🤔

At least in the network inspector I cannot find a request that times out. The request of `extensions/sources/index.js` with its 300 kB finishes with a 200. 

When throttled, the loading of the extensions source take almost the whole app initialization duration. Could this maybe raise some timing issues?

_Full speed:_
App initialization: 80ms
Extensions index.js: 23 ms

_Throttled:_
App initialization: 552 ms
Extensions index.js: 503 ms

BTW, when we run the Admin app on the development server the issue does not appear. Maybe because the extensions aren't bundled here!?

**Comment by christianrr** [comment]
I was able to reproduce the issue locally with a 700kb module, which helped to further investigate the issue.

Basically, there are two issues:
1. The `beforeEach` navigation guard in `router.ts` redirects to `to.fullPath`, whenever the app is not hydrated yet. However if `loadExtensions` takes longer, routes are not registered yet and the redirect navigates to 404 Page not found. 
2. The callback for registering the routes is registered after `registerExtensions` ends. If loading of extensions takes longer, the call of `onHydrateExtensions` in `hydrate.ts` happens even before callbacks are registered in `onHydrateCallbacks`. Therefore, routes get never instantiated. 

What we found is that if `app.use(router)` happens after `registerExtensions`, everything works as expected since app hydration and initial routing happen after `registerExtensions`.

@rijkvanzanten Could this be a potential fix, or is there any downside of changing the initialization order in `main.ts`?

```
	app.use(i18n);
	app.use(createPinia());
	app.use(createHead());

	...

	await loadExtensions();
	registerExtensions(app);
	app.use(router);
```
